src/Sentry/Protocol/Envelopes/Envelope.cs

```diff
@@ -325,6 +326,35 @@ public static Envelope FromTransaction(Transaction transaction)
         return new Envelope(eventId, header, items);
     }
 
+    /// <summary>
+    /// Creates an envelope that contains one or more <see cref="CodeLocations"/>
+    /// </summary>
+    internal static Envelope FromCodeLocations(CodeLocations codeLocations)
+    {
+        var header = DefaultHeader;
+
+        List<EnvelopeItem> items = new();
+        items.Add(EnvelopeItem.FromCodeLocations(codeLocations));
```

Comment:
Does `Add` only allocates an array with 1 element? I believe adding the first item u can:

var items = new List<EnvelopeItem>(1) {EnvelopeItem.FromCodeLocations(};

Though it's anyway calling `Add` just a syntactical sugar. But the `(1)` in there already gives you the list in the size we'll use.